### PR TITLE
Authenticating user

### PR DESCRIPTION
**Pull request by Gilbertelnino**
**What does this PR do?**

- Authenticating user using jwt

**How to run it manually**

- Pull the latest changes
- Run npm install
- Run `npm run dev`

### CODE REVIEW

File: src/router/user.routes.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import express from "express";
+import verify from "../middlewares/verifyToken";
+import { UserController } from "../controllers/UserController";
+const router = express.Router();
+
+router.post("/signup", UserController.createUser);
+
+router.post("/login", UserController.loginUser);
+router.delete("/:id", verify, UserController.deleteUser);
```
Comment (niyodusengaclement) [inline src/router/user.routes.js]: use understandable routes like "/user/:id"

File: src/controllers/UserController.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    // check if password is correct
+    const validPassword = await bcrypt.compare(
+      req.body.password,
+      user.password
+    );
+    if (!validPassword) return onError(res, 401, "Invalid Email or Password");
+
+    // create a token
+
+    const token = jwt.sign({ user: user }, process.env.JWT_SECRET);
```
Comment (niyodusengaclement) [inline src/controllers/UserController.js]: Do not expose all of the user information on the token, you can see that even password is in the user object u are signing

**Review by niyodusengaclement — CHANGES_REQUESTED** [review]
Fix the issues

File: src/router/user.routes.js (hunk 1 above)
Comment (Gilbertelnino) [reply src/router/user.routes.js]: it is "/user/:id" currently the reason why you don't see it here it is because I set /user as base router in index.js

**Pull request closed without merging**